CONTRIBUTING.md

```diff
@@ -50,9 +46,1148 @@ Contributing features, documentation, tests
 * Please let us know if you plan to work on a ticket. This way we can make sure
   redundant work is avoided.
 
+# Main sections
 
-Additional Resources
---------------------
+- [Source Code](#source-code)
+- [Building](#building)
+- [Running](#running)
+- [Debugging](#debugging)
+  - [Linux Core Dumps](#linux-core-dumps)
+  - [Windows Core Dumps](#windows-core-dumps)
+- [Unittests](#unittests)
+  - [invoking driver tests](#driver-tests) via scripts/unittests
+  - [capturing test HTTP communication](#running-tcpdump--windump-for-the-sut) but
+    [forcing communication to use plain-text JSON](#forcing-downgrade-from-vpack-to-json)
+  - [Evaluating previous testruns](#evaluating-json-test-reports-from-previous-testruns)
+    sorting by setup time etc. 
+- [what to test where and how](tests/README.md)
+
+## Source Code
+
+### Git
+
+Setting up git for automatically merging certain automatically generated files
+in the ArangoDB source tree:
+
+    git config --global merge.ours.driver true
+
+### Style Guide
+
+We use `clang-format` to enfore consistent code formatting. Check STYLEGUIDE.md
```

Comment:
Should we still reference the .clang-format file here directly? We use `clang-format` with the following [style](link).